src/main/java/cloud/fogbow/ras/core/plugins/interoperability/cloudstack/attachment/v4_9/CloudStackAttachmentPlugin.java

```diff
@@ -52,115 +49,153 @@ public boolean hasFailed(String cloudState) {
+        CloudStackUrlUtil.sign(uriRequest, cloudStackUser.getToken());
+
+        try {
+            String jsonResponse = CloudStackCloudUtils.doRequest(
+                    this.client, uriRequest.toString(), cloudStackUser);
+            AttachVolumeResponse response = AttachVolumeResponse.fromJson(jsonResponse);
+            return response.getJobId();
+        } catch (HttpResponseException e) {
+            throw CloudStackHttpToFogbowExceptionMapper.get(e);
         }
-        
-        AttachmentJobStatusResponse response = AttachmentJobStatusResponse.fromJson(jsonResponse);
-        
-        return loadInstanceByJobStatus(order.getInstanceId(), response);
     }
 
-    private AttachmentInstance loadInstanceByJobStatus(String attachmentInstanceId,
-                                       AttachmentJobStatusResponse response) throws UnexpectedException {
+    @NotNull
+    @VisibleForTesting
+    AttachmentInstance loadInstanceByJobStatus(String attachmentInstanceId,
+                                               @NotNull AttachmentJobStatusResponse response)
+            throws UnexpectedException {
         
         int status = response.getJobStatus();
         switch (status) {
-            case JOB_STATUS_PENDING:
-                return new AttachmentInstance(attachmentInstanceId, PENDING_STATE, null, null, null);
-            case JOB_STATUS_COMPLETE:
+            case CloudStackCloudUtils.JOB_STATUS_PENDING:
+                return mountInstance(attachmentInstanceId, CloudStackCloudUtils.PENDING_STATE);
+            case CloudStackCloudUtils.JOB_STATUS_COMPLETE:
                 AttachmentJobStatusResponse.Volume volume = response.getVolume();
-                return mountInstance(volume);
-            case JOB_STATUS_FAILURE:
-                return new AttachmentInstance(attachmentInstanceId, FAILURE_STATE, null, null, null);
+                return mountCompleteInstance(volume);
+            case CloudStackCloudUtils.JOB_STATUS_FAILURE:
+                logFailure(response);
+                return mountInstance(attachmentInstanceId, CloudStackCloudUtils.FAILURE_STATE);
             default:
-                throw new UnexpectedException();
+                throw new UnexpectedException(Messages.Error.UNEXPECTED_JOB_STATUS);
         }
     }
 
-    private AttachmentInstance mountInstance(AttachmentJobStatusResponse.Volume volume) {
+    @VisibleForTesting
+    void logFailure(@NotNull AttachmentJobStatusResponse response) throws UnexpectedException {
+        CloudStackErrorResponse errorResponse = response.getErrorResponse();
+        String errorText = String.format(FAILED_ATTACH_ERROR_MESSAGE,
+                errorResponse.getErrorCode(), errorResponse.getErrorText());
+        LOGGER.error(String.format(Messages.Error.ERROR_WHILE_ATTACHING_VOLUME_GENERAL, errorText));
+    }
+
+    @NotNull
+    private AttachmentInstance mountCompleteInstance(@NotNull AttachmentJobStatusResponse.Volume volume) {
```

Comment:
Rename the _mountCompleteInstance_ method to _buildCompleteInstance_.

Reply:
done